### PR TITLE
Add SessionConcurrentSpec

### PR DESCRIPTION
**Pull request by geoffjohn11**

### CODE REVIEW

**Comment by geoffjohn11** [comment]
@BalmungSan This is an example test of sharing a session in parallel with the cats IO effect type.  Not sure we should merge this, but I wanted to share this example.

**Comment by BalmungSan** [comment]
Cool @geoffjohn11 I will try to pull and play with this code.

**Comment by BalmungSan** [comment]
@dimafeng  This PR Adds a new spec that checks the behavior of using a session concurrently.
For now, it fails. But ideally, we would change this in the future, to ensure the session can be sued concurrently for all effect types _(see #44 )_.

@geoffjohn11  Can you take a look at the changes?

File: core/src/test/scala/neotypes/FutureConcurrentSessionSpec.scala [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package neotypes
+
+import scala.concurrent.Future
+
+class FutureConcurrentSessionSpec extends ConcurrentSessionSpec[Future] {
+  override def fToFuture[T](future: Future[T]): Future[T] =
+    future
+
+  override def runConcurrently(a: Future[Unit], b: Future[Unit]): Future[Unit] =
+    for (_ <- a; _ <- b) yield () // Because Futures are eager, they are already running concurrently.
```
Comment (geoffjohn11) [inline core/src/test/scala/neotypes/FutureConcurrentSessionSpec.scala]: This comment is helpful

File: core/src/test/scala/neotypes/ConcurrentSessionSpec.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package neotypes
+
+import neotypes.implicits.mappers.all._
+import neotypes.implicits.syntax.cypher._
+import neotypes.internal.syntax.async._
+import org.neo4j.driver.v1.exceptions.ClientException
+import org.scalatest.compatible.Assertion
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+
```
Comment (geoffjohn11) [inline core/src/test/scala/neotypes/ConcurrentSessionSpec.scala]: Intellij tells me that you can clean up the imports.

Comment (BalmungSan) [reply core/src/test/scala/neotypes/ConcurrentSessionSpec.scala]: Will take a look :+1:

**Pull request merged**